isis/tests/FunctionalTestsMappt.cpp

```diff
@@ -0,0 +1,173 @@
+#include <QTextStream>
+#include <QStringList>
+#include <QFile>
+#include <QDebug>
+
+#include "mappt.h"
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "SpecialPixel.h"
+
+#include "gmock/gmock.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/mappt.xml").expanded();
+
+TEST_F(DefaultCube, FunctionalTestMapptImageTest) {
+  QVector<QString> args = {"append=false", "type=image", "sample=1", "line=1"};
+  UserInterface options(APP_XML, args);
+  Pvl appLog;
+
+  mappt(projTestCube, options, &appLog);
+  PvlGroup mapPoint = appLog.findGroup("Results");
+  
+  std::cout << mapPoint << std::endl;
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, mapPoint.findKeyword("FilterName"), "CLEAR");
+  EXPECT_EQ( (double) mapPoint.findKeyword("Band"), 1);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetographicLatitude"), 9.3870849567571);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetocentricLatitude"), 9.2788326719634);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest360Longitude"), 359.14528612684);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast360Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast180Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest180Longitude"), -0.85471387315751); 
+}
+
+TEST_F(DefaultCube, FunctionalTestMapptGroundTest) {
+  QVector<QString> args = {"append=false", "type=ground", "latitude=9.2788326719634", "longitude=0.85471387315749"};
+  UserInterface options(APP_XML, args);
+  Pvl appLog;
+  mappt(projTestCube, options, &appLog);
+  
+  PvlGroup mapPoint = appLog.findGroup("Results");
+
+  std::cout << mapPoint << std::endl;
+
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, mapPoint.findKeyword("FilterName"), "CLEAR");
+  EXPECT_EQ( (double) mapPoint.findKeyword("Band"), 1);
+  EXPECT_NEAR( (double) mapPoint.findKeyword("Sample"), 1, 1e-8);
+  EXPECT_NEAR( (double) mapPoint.findKeyword("Line"), 1, 1e-8);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetographicLatitude"), 9.3870849567571);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PlanetocentricLatitude"), 9.2788326719634);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest360Longitude"), 359.14528612684);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast360Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveEast180Longitude"), 0.85471387315749);
+  EXPECT_DOUBLE_EQ( (double) mapPoint.findKeyword("PositiveWest180Longitude"), -0.85471387315751); 
```

Comment:
This needs to check that the output contains the correct X and Y projection values.